docs/user/explanations/system-simulation-adapters.rst

```diff
@@ -0,0 +1,154 @@
+System Simulation Adapters
+==========================
+
+Adapters may also be used with system simulations (`SystemSimulationComponent`),
+as well as with devices (`DeviceSimulation`). This would allow you, for example, to
+query what device components are inside the system simulation component, and how they
+are wired together.
+
+Currently system adapters need both wiring and components. There is a helper base class
+for implementing this.  
+
+.. code-block:: python
+        
+    class BaseSystemSimulationAdapter:
+        """A base for a SystemSimulationComponent adapter."""
+
+        _components: Dict[ComponentID, Component]
+        _wiring: Union[Wiring, InverseWiring]
+
+        def setup_adapter(
+            self,
+            components: Dict[ComponentID, Component],
+            wiring: Union[Wiring, InverseWiring],
+        ) -> None:
+            """Provides the components and wiring of a SystemSimulationComponent."""
+            self._components = components
+            self._wiring = wiring
+
+
+
+Nested Amplifier Example
+------------------------
+
+The following system adapter is a simple `CommandAdapter`. This Can be used to query the
+SystemSimulationComponent for a list of ID's of the components it contains; and to
+return the wiring map of the components.
+
+.. code-block:: python
+
+    class SystemSimulationAdapter(BaseSystemSimulationAdapter, CommandAdapter):
+
+        _byte_format: ByteFormat = ByteFormat(b"%b\r\n")
+
+        @RegexCommand(r"ids", False, "utf-8")
+        async def get_component_ids(self) -> bytes:
+            """Returns a list of ids for all the components in the system simulation."""
+            return str(self._components.keys()).encode("utf-8")
+
+        @RegexCommand(r"wiring", False, "utf-8")
+        async def get_wiring(self) -> bytes:
+            """Returns the wiring object used by the nested scheduler."""
+            return str(self._wiring).encode("utf-8")
+
+To use this adapter we would need to put it in an adapter container with TcpIo and
+assign it as an argument in the SystemSimulationComponent.
+
+.. code-block:: python
+
+    @pydantic.v1.dataclasses.dataclass
+    class NestedAmplifierWithAdapter(ComponentConfig):
+        """Simulation of a nested amplifier with a CommandAdapter."""
+
+        name: ComponentID
+        inputs: Dict[PortID, ComponentPort]
```

Comment:
How does repeating a field name on a Pydantic dataclass work? Is this just to be explicit in the docs?

Reply:
I just copied the other one and it works so honestly I don't know